scripts/tests/twister/test_testinstance.py

```diff
@@ -56,22 +56,30 @@ def test_check_build_or_run(class_testplan, monkeypatch, all_testsuites_dict, pl
-    (True, False, True, ["demo_board_2"], "native", '\nCONFIG_COVERAGE=y\nCONFIG_COVERAGE_DUMP=y\nCONFIG_ASAN=y'),
-    (False, False, True, ["demo_board_2"], 'native', '\nCONFIG_COVERAGE=y\nCONFIG_COVERAGE_DUMP=y'),
-    (True, False, True, ["demo_board_2"], 'mcu', '\nCONFIG_COVERAGE=y\nCONFIG_COVERAGE_DUMP=y'),
-    (False, False, False, ["demo_board_2"], 'native', ''),
-    (False, False, True, ['demo_board_1'], 'native', ''),
-    (True, False, False, ["demo_board_2"], 'native', '\nCONFIG_ASAN=y'),
-    (False, True, False, ["demo_board_2"], 'native', '\nCONFIG_UBSAN=y'),
+    (True, True, True, ["demo_board_2"], "native", None, '\nCONFIG_COVERAGE=y\nCONFIG_COVERAGE_DUMP=y\nCONFIG_ASAN=y\nCONFIG_UBSAN=y'),
+    (True, False, True, ["demo_board_2"], "native", None, '\nCONFIG_COVERAGE=y\nCONFIG_COVERAGE_DUMP=y\nCONFIG_ASAN=y'),
+    (False, False, True, ["demo_board_2"], 'native', None, '\nCONFIG_COVERAGE=y\nCONFIG_COVERAGE_DUMP=y'),
+    (True, False, True, ["demo_board_2"], 'mcu', None, '\nCONFIG_COVERAGE=y\nCONFIG_COVERAGE_DUMP=y'),
+    (False, False, False, ["demo_board_2"], 'native', None, ''),
+    (False, False, True, ['demo_board_1'], 'native', None, ''),
+    (True, False, False, ["demo_board_2"], 'native', None, '\nCONFIG_ASAN=y'),
+    (False, True, False, ["demo_board_2"], 'native', None, '\nCONFIG_UBSAN=y'),
+    (False, False, False, ["demo_board_2"], 'native', ["CONFIG_NULL=y"], 'CONFIG_NULL=y'),
+    (False, False, False, ["demo_board_2"], 'native', ["arch:x86_demo:CONFIG_NULL=y"], 'CONFIG_NULL=y'),
+    (False, False, False, ["demo_board_2"], 'native', ["arch:arm_demo:CONFIG_NULL=y"], ''),
+    (False, False, False, ["demo_board_2"], 'native', ["platform:demo_board_2:CONFIG_NULL=y"], 'CONFIG_NULL=y'),
+    (False, False, False, ["demo_board_2"], 'native', ["platform:demo_board_1:CONFIG_NULL=y"], ''),
 ]
 
-@pytest.mark.parametrize("enable_asan, enable_ubsan, enable_coverage, coverage_platform, platform_type, expected_content", TESTDATA_2)
-def test_create_overlay(class_testplan, all_testsuites_dict, platforms_list, enable_asan, enable_ubsan, enable_coverage, coverage_platform, platform_type, expected_content):
-    """Test correct content is written to testcase_extra.conf based on if conditions
-    TO DO: Add extra_configs to the input list"""
+@pytest.mark.parametrize("enable_asan, enable_ubsan, enable_coverage, coverage_platform, platform_type, extra_configs, expected_content", TESTDATA_2)
+def test_create_overlay(class_testplan, all_testsuites_dict, platforms_list, enable_asan, enable_ubsan, enable_coverage, coverage_platform, platform_type, extra_configs, expected_content):
+    """Test correct content is written to testcase_extra.conf based on if conditions."""
     class_testplan.testsuites = all_testsuites_dict
     testcase = class_testplan.testsuites.get('scripts/tests/twister/test_data/testsuites/samples/test_app/sample_test.app')
+
+    if extra_configs:
+        testcase.extra_configs = extra_configs
```

Comment:
@LukaszMrugala please split the `extra_configs` fix and the typo fix into separate commits and update the PR's description that there are two fixes here.